### PR TITLE
feat(sdk): Filter DataPicker models for `CreateQuestion`

### PR DESCRIPTION
**Pull request by oisincoveney**
Adds filtering for the DataPicker by passing the `models` prop from the Notebook component into the `DataStep` component.

In terms of why this PR has a lot of files, I wanted to ensure that the props are accurately reflected in each "passing down" of the data, so that's why there's quite a few `Pick` statements and things like that. They won't affect the functionality at all, but if we do decide to add a new type to the filter, then we will immediately know if there's a type issue

### CODE REVIEW

File: enterprise/frontend/src/embedding-sdk/components/private/InteractiveQuestion/components/Notebook.tsx [hunk 1]
```diff
@@ -7,17 +7,38 @@ import {
...
+  metric: "metric",
+  model: "dataset",
+  question: "card",
+  table: "table",
+} as const;
+
+const getNotebookModelFilter = (
+  models: NotebookProps["models"],
+): QBNotebookProps["models"] =>
+  models.map(model => notebookModelFilterMap[model]);
```
Comment (oisincoveney) [inline enterprise/frontend/src/embedding-sdk/components/private/InteractiveQuestion/components/Notebook.tsx:28]: will fix this tomorrow

**Comment by oisincoveney** [comment]
@metabase/core-frontend-querying - Since this is more or less in your realm, it might make more sense for y'all to review this. Basically, the only thing SDK-related in this code is passing the `models` prop to the QB notebook

**Review by ranquild — CHANGES_REQUESTED** [review]
Should we consider adding a context? I don't think we should drill down props like this.

Btw - the types are off in terms of what type depends on which, but this would be reverted if we add a context.

**Comment by oisincoveney** [comment]
> Should we consider adding a context? I don't think we should drill down props like this.
> 
> Btw - the types are off in terms of what type depends on which, but this would be reverted if we add a context.

totally up for a context 😄 I didn't want to get to "context-happy" for this change, but if you also think it makes sense then let's do it

**Review by ranquild — APPROVED** [review]
I was thinking about slightly different behavior. What if we:
1. Add a default value for `modelsFilterList` in `Notebook` and make it `["table", "card", "dataset", "metric"]`
2. Change `NotebookDataPicker` to use `modelsFilterList` and filter out `"metric"` if `useMetric` is false?

The thing is that, when you allow metrics in the SDK, we should still remove metrics from joins. 

Approving in advance to save time. Hopefully this is a straightforward change.

**Comment by oisincoveney** [comment]
> 1. ["table", "card", "dataset", "metric"]

Ah, I see what you mean - instead of including them when we have 'hasMetrics=true' we should just exclude them when 'hasMetrics=false'?

If that's the case then I've pushed that change - I'll merge it after one final confirmation later today when you're awake 😄

File: enterprise/frontend/src/embedding-sdk/components/private/InteractiveQuestion/context/InteractiveQuestionProvider.tsx [hunk 2]
```diff
@@ -26,6 +28,19 @@ export const InteractiveQuestionContext = createContext<
 
 const DEFAULT_OPTIONS = {};
 
+const FILTER_MODEL_MAP: Record<EntityTypeFilterKeys, DataPickerValue["model"]> =
```
Comment (oisincoveney) [inline enterprise/frontend/src/embedding-sdk/components/private/InteractiveQuestion/context/InteractiveQuestionProvider.tsx:31]: I've added this to provide some the user-facing terms to the user while still using our terms internally

File: frontend/src/metabase-types/api/activity.ts [hunk 3]
```diff
@@ -45,11 +47,7 @@ export interface RecentCollectionItem extends BaseRecentItem {
   model: "collection" | "dashboard" | "card" | "dataset" | "metric";
   can_write: boolean;
   database_id?: DatabaseId; // for models and questions
-  parent_collection: {
-    id: number | null;
-    name: string;
-    authority_level?: "official" | null;
-  };
+  parent_collection: Pick<Collection, "id" | "name" | "authority_level">;
```
Comment (oisincoveney) [inline frontend/src/metabase-types/api/activity.ts:50]: I noticed that in practice we're using the same data types so I figured we should do this

File: frontend/src/metabase/palette/components/PaletteResults.unit.spec.tsx [hunk 4]
```diff
@@ -86,7 +86,7 @@ const recents_1 = createMockRecentCollectionItem({
   model: "dataset",
   moderated_status: "verified",
   parent_collection: {
-    id: null,
+    id: "root",
```
Comment (oisincoveney) [inline frontend/src/metabase/palette/components/PaletteResults.unit.spec.tsx:89]: this just fixes a type error on the unit test, it doesn't affect anything

File: frontend/src/metabase/querying/notebook/components/JoinStep/JoinStep.unit.spec.tsx [hunk 5]
```diff
@@ -187,16 +188,18 @@ function setup({
...
-        step={step}
-        stageIndex={step.stageIndex}
-        query={query}
-        color="brand"
-        isLastOpened={false}
-        readOnly={readOnly}
-        reportTimezone="UTC"
-        updateQuery={onChange}
-      />
+      <NotebookProvider>
```
Comment (oisincoveney) [inline frontend/src/metabase/querying/notebook/components/JoinStep/JoinStep.unit.spec.tsx:191]: fixes a test error that complains about not having the context available

File: frontend/src/metabase/querying/notebook/components/NotebookDataPicker/NotebookDataPicker.tsx [hunk 6]
```diff
@@ -44,10 +45,15 @@ export function NotebookDataPicker({
   table,
   databaseId,
   placeholder = title,
-  hasMetrics,
+  hasMetrics = false,
```
Comment (oisincoveney) [inline frontend/src/metabase/querying/notebook/components/NotebookDataPicker/NotebookDataPicker.tsx:48]: changes the type within the component from `boolean | undefined` to `boolean` just to make things a little easier

File: frontend/src/metabase/querying/notebook/components/NotebookDataPicker/NotebookDataPicker.tsx [hunk 7]
```diff
@@ -44,10 +45,15 @@ export function NotebookDataPicker({
...
   isDisabled,
   onChange,
 }: NotebookDataPickerProps) {
+  const { modelsFilterList } = useNotebookContext();
```
Comment (oisincoveney) [inline frontend/src/metabase/querying/notebook/components/NotebookDataPicker/NotebookDataPicker.tsx:52]: Now, when `hasMetrics` is `false`, we exclude `metric`.

The default behavior is that we have `metric` in our `modelsFilterList` array, so if `hasMetrics` is true, `metric` will be there

File: frontend/src/metabase/querying/notebook/components/NotebookStep/NotebookStep.unit.spec.tsx [hunk 8]
```diff
@@ -31,14 +32,16 @@ function setup({ step = createMockNotebookStep() }: SetupOpts = {}) {
...
   renderWithProviders(
-    <NotebookStep
-      step={step}
-      isLastStep={false}
-      isLastOpened={false}
-      reportTimezone="Europe/London"
-      openStep={openStep}
-      updateQuery={updateQuery}
-    />,
+    <NotebookProvider>
```
Comment (oisincoveney) [inline frontend/src/metabase/querying/notebook/components/NotebookStep/NotebookStep.unit.spec.tsx:35]: fixes a test error that complains about not having the context available

File: enterprise/frontend/src/embedding-sdk/README.md [hunk 9]
```diff
@@ -110,11 +110,11 @@ const metabaseConfig = {
 ### JWT Authentication
 
 1. Go to Admin settings > Authentication > JWT
```
Comment (npretto) [inline enterprise/frontend/src/embedding-sdk/README.md:112]: for the next time, when you see that a file needs to be formatted, can you do that in a separate PR?
This pr has been open for 3 weeks, we're lucky we didn't get conflicts

File: enterprise/frontend/src/embedding-sdk/components/private/InteractiveQuestion/context/InteractiveQuestionProvider.tsx (hunk 2 above)
Comment (npretto) [reply enterprise/frontend/src/embedding-sdk/components/private/InteractiveQuestion/context/InteractiveQuestionProvider.tsx:31]: nit: the name of the const doesn't really give me the direction of the mapping

File: enterprise/frontend/src/embedding-sdk/README.md (hunk 9 above)
Comment (oisincoveney) [reply enterprise/frontend/src/embedding-sdk/README.md:112]: crap, I thought the diff was large because I wrote a decent amount... 🙃

Comment (oisincoveney) [reply enterprise/frontend/src/embedding-sdk/README.md:112]: nice catch. I'll fix that before merging

File: frontend/src/metabase/querying/notebook/components/Notebook/context.tsx [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+
+type NotebookContextType = {
+  modelsFilterList: DataPickerValue["model"][];
+};
+
+export const NotebookContext = createContext<NotebookContextType | undefined>(
+  undefined,
+);
+
+export const NotebookProvider = ({
```
Comment (npretto) [inline frontend/src/metabase/querying/notebook/components/Notebook/context.tsx:13]: Can't this be `NotebookProvider = NotebookContext.Provider`?

Comment (oisincoveney) [reply frontend/src/metabase/querying/notebook/components/Notebook/context.tsx:13]: We would have to initialize the NotebookProvider with 

<NotebookProvider value={{ modelsFilterList: [...] }}>
    {... whatever ...}
</NotebookProvider>


like we do here, but for all of the instances of the NotebookProvider. I would prefer to keep the props the same as other components

**Review by npretto — APPROVED** [review]

**Pull request merged**